### ISSUE TITLE
Wrong URL for Azure Gov in ProjectNamiBlobCacheCleanup.exe

### ISSUE THREAD

**Issue by readej** [ProjectNami/ProjectNamiBlobCacheCleanup#1]
When the WebJob runs I get an error that states that the remote name could not be resolved: '<name-of-storage-account>.blob.core.windows.net' at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext operationContext)

The url .blob.core.windows.net is incorrect for Azure Government. It should be .blob.core.usgovcloudapi.net

Is there a way to easily fix this?

**Comment by patrickebates** [comment]
That's strange.  We aren't setting that anywhere within the code, so I can only assume it's related to the Azure Storage library from Microsoft.  I can look at updating it to a newer version and see if that corrects the issue.

But I suspect you are generating more errors elsewhere.  The Blob Cache IIS module uses the same or similar version of the Azure Storage library.  I wouldn't be surprised if your cache serving is broken as well.

**Comment by readej** [comment]
Yes, I had to change some settings in the Resource.php in \library\WindowsAzure\Common\Internal folder for the blob cache plugin.

I believe I got the cache working after replacing '.core.windows.net' with 'core.usgovcloudapi.net'

I will look into the Azure Storage library and see if I can solve this issue there. Thanks!

**Comment by patrickebates** [comment]
The plugin only writes to the cache, the IIS Module reads from the cache and serves the hits.  So while the plugin may be working after your edits, I suspect that either the IIS Module is failing as well or you haven't installed and configured it.

**Comment by readej** [comment]
How would I be able to tell if the IIS Module is failing?
I installed the plugin and Cache Cleanup following the instructions in the readme.txt
I did not install the Cache Loader because I wanted to get Cleanup working first.

**Comment by patrickebates** [comment]
Any entries in the site log file folder, other than PHP errors?

**Comment by patrickebates** [comment]
I now know what changes need to be made to support the various Azure Gov options, but now I find my Visual Studio locked out of updating these GitHub repos...

Once I figure out that problem, you can expect changes to fix this issue.

**Comment by patrickebates** [comment]
Just uploaded a change to the Blob Cache IIS module which supports a new app setting to specify the connection root.  I will try to test it and provided a compiled file for you to test soon.

Can you confirm that the change you made to the plugin was updating the BLOB_BASE_DNS_NAME in Resources.php?
